content/blog/2019/07/2019-07-04-use-casc-for-exws.adoc

```diff
@@ -0,0 +1,141 @@
+---
+layout: post
+title: "Use Terraform with External Workspace Manager to Support EFS"
+tags:
+- jenkins
+- gsoc
+author: yufei_zhang
+description: "A brief example of using Terraform with EFS" # optional
+opengraph:
+---
+
+=== Why you want to use External Workspace Manager with EFS
+
+What is Terraform : A CasC software, like CloudFormation and Ansible designed for managing cloud resources. It can free you from managing cloud resource by console, and can be version controlled using git as it is configuration-as-code. Currently it supports AWS and Azure best. AWS CloudFormation is for AWS only.
+What is extenal-workspace-manager : It is a pipeline plugin which customizes the workspace path. The same disk can be mounted to multiple nodes to share resources
+What is EFS : Amazon Elastic File System, an Network File System, can be shared among a lot of EC2 instances.
+
+Who should read this blog :
+
+. build Jenkins based on cloud, want to have good version control over all resources including infrastructure; 
+. You want to share workspace among agents (slaves), there might be an upstream job creating a large file and don't want to copy it into downstream job's workspace because it's time-consuming; 
```

Comment:
```suggestion
. You want to share workspace among agents, there might be an upstream job creating a large file and don't want to copy it into downstream job's workspace because it's time-consuming; 
```

Please avoid using old terminology